### PR TITLE
DAOS-7446 test: Run container/filesystem copy under fault injection.

### PR DESCRIPTION
**Pull request by ashleypittman**
Add a test to NLT to run 'daos filesystem copy' under fault
injection.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
This PR looks to be hitting at least one assert() in src/utils/daos_hdlr.c that should be D_ASSERT()

File: src/common/proc.c [hunk 1]
```diff
@@ -177,7 +177,8 @@ crt_proc_daos_prop_t(crt_proc_t proc, crt_proc_op_t proc_op, daos_prop_t **data)
 		prop->dpp_reserv = tmp;
 		rc = crt_proc_prop_entries(proc, proc_op, prop);
 		if (rc) {
-			daos_prop_free(prop);
+			D_FREE(prop->dpp_entries);
+			D_FREE(prop);
```
Comment (ashleypittman) [inline src/common/proc.c:181]: This change fixes the crash and will want backporting.

Comment (dsikich) [reply src/common/proc.c:181]: @ashleypittman I guess I am not totally sure why this fixes it either... I am not really familiar with this portion of the codebase either. Maybe someone who has worked in this part of the code would know better?

Comment (ashleypittman) [reply src/common/proc.c:181]: @dmiter @liuxuezhao any ideas?  Without this change there's a crash on free, but this change avoids the issue.

Comment (dsikich) [reply src/common/proc.c:181]: @ashleypittman maybe there is some difference in daos_prop_free vs D_FREE? I am having trouble finding the implementation of daos_prop_free.

Comment (ashleypittman) [reply src/common/proc.c:181]: https://github.com/daos-stack/daos/blob/master/src/common/prop.c#L94

The error that this changes fixes is this one, https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5655/8/valgrindResult/pid=61934,0x0/ bit I can't see how dpe_val_ptr would ever be set.

File: src/client/dfs/dfs.c [hunk 2]
```diff
@@ -1281,7 +1282,7 @@ dfs_cont_create(daos_handle_t poh, uuid_t co_uuid, dfs_attr_t *attr,
 	 */
 	rc = insert_entry(super_oh, DAOS_TX_NONE, "/", 1, &entry);
 	if (rc && rc != EEXIST) {
-		D_ERROR("Failed to insert root entry (%d).", rc);
+		D_ERROR("Failed to insert root entry, "DF_RC"\n", DP_RC(rc));
```
Comment (dmiter) [inline src/client/dfs/dfs.c:1285]: This is incorrect. **rc** here is errno.

**Comment by ashleypittman** [comment]
I've disabled the test from running in CI as it's incomplete, however I'd like to land this PR as-is because it contains a number of valuable fixes.  Can I ask for fresh reviews please?

File: src/client/dfs/dfs.c [hunk 3]
```diff
@@ -1443,7 +1443,8 @@ dfs_mount(daos_handle_t poh, daos_handle_t coh, int flags, dfs_t **_dfs)
 		/** Set gid to nobody */
 		rc = daos_acl_principal_to_gid("nobody@", &dfs->gid);
 	if (rc) {
-		D_ERROR("Unable to convert owner to gid\n");
+		D_ERROR("Unable to convert owner to gid "DF_RC"\n",
```
Comment (daltonbohning) [inline src/client/dfs/dfs.c:1446]: Similar
```suggestion
		D_ERROR("Unable to convert owner to gid, "DF_RC"\n",
```

File: src/common/prop.c [hunk 4]
```diff
@@ -85,7 +83,6 @@ daos_prop_fini(daos_prop_t *prop)
 	}
 
 	D_FREE(prop->dpp_entries);
-	prop->dpp_entries = NULL;
```
Comment (daltonbohning) [inline src/common/prop.c:88]: Is it safe to remove this? What if there is code that relies on this being set to NULL?
I.e. D_FREE free's this, but does not set to NULL, correct?

File: src/client/array/dc_array.c [hunk 5]
```diff
@@ -1735,8 +1733,8 @@ dc_array_io(daos_handle_t array_oh, daos_handle_t th,
 					      tse_task2sched(task),
 					      0, NULL, &io_task);
 			if (rc != 0) {
-				D_ERROR("Update dkey "DF_U64" failed (%d)\n",
-					params->dkey_val, rc);
+				D_ERROR("Update dkey "DF_U64" failed "DF_RC"\n",
+					params->dkey_val, DP_RC(rc));
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/client/array/dc_array.c:1737]: No, it's a uint64_t

File: src/client/dfs/dfs.c [hunk 6]
```diff
@@ -1432,7 +1431,8 @@ dfs_mount(daos_handle_t poh, daos_handle_t coh, int flags, dfs_t **_dfs)
 		/** Set uid to nobody */
 		rc = daos_acl_principal_to_uid("nobody@", &dfs->uid);
 	if (rc) {
-		D_ERROR("Unable to convert owner to uid\n");
+		D_ERROR("Unable to convert owner to uid "DF_RC"\n",
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/client/dfs/dfs.c:1434]: Hard to say, I'm not aware of a standard.  I'm pushing to fix the worst transgressions but that's causing enough churn without dealing with the minor issues as well.  Once we've got better tools or more fault injection coverage may be we can re-visit this.

What I'd really like is a D_ERROR() style macro that took a descriptor pointer and a rc value and did the formatting etc automatically.

File: src/common/prop.c (hunk 4 above)
Comment (ashleypittman) [reply src/common/prop.c:88]: D_FREE() does set things to NULL.

Comment (daltonbohning) [reply src/common/prop.c:88]: Oh? Good to know

**Review by daltonbohning — APPROVED** [review]

File: src/client/dfs/dfs.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/client/dfs/dfs.c:1434]: Another thing is the fault injection code is *really good* at finding bugs, some serious, some cosmetic.  What I'd like to do is to fix the serious ones (there's already been one ticket and PR landed as a side effect of this PR) quickly, improve cosmetic ones so they're useful/correct, then move on to adding more coverage.

**Review by daltonbohning — APPROVED** [review]

**Review by dsikich — APPROVED** [review]

**Pull request merged**